File: Java/src/main/java/com/gildedrose/items/Conjured.java
package com.gildedrose.items;

import com.gildedrose.Item;

/**
 * - Quality of Conjured decreases over time
 * - It decreases twice as fast as Standard items
 * - If expired, it decreases twice as fast
 */
class Conjured extends CustomItem {

    private static final int MULTIPLICATOR = 2;

    @Override
    public void updateQuality(Item item) {
        decreaseQualityByMultiplicator(item, MULTIPLICATOR);

        if (isExpired(item.sellIn)) decreaseQualityByMultiplicator(item, MULTIPLICATOR);

        super.updateQuality(item);
    }
}

File: Kotlin/src/main/kotlin/com/gildedrose/items/Conjured.kt
package com.gildedrose.items

import com.gildedrose.Item

private const val MULTIPLICATOR = 2

/**
 * - Quality of Conjured decreases over time
 * - It decreases twice as fast as Standard items
 * - If expired, it decreases twice as fast
 */
class Conjured : CustomItem {
    override fun updateQuality(item: Item) {
        decreaseQualityByMultiplicator(item, MULTIPLICATOR)

        if (isExpired(item.sellIn)) decreaseQualityByMultiplicator(item, MULTIPLICATOR)

        super.updateQuality(item)
    }
}
